### PR TITLE
Add target doxy to build developers documentation

### PR DESCRIPTION
**Pull request by estebanzimanyi**
Enable building the developers documentation with
```
make doxy
```
Contrary to pgRouting, this is independent of building the users documentation.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,58 +44,58 @@ if(NOT WIN32)
...
+
 # Currently does not work with other versions of PostGIS
 find_package(POSTGIS 2.5 EXACT REQUIRED)
 add_definitions(-DPOSTGIS_VERSION_STR="${POSTGIS_VERSION_STR}")
 add_definitions(-DPOSTGIS_VERSION_NUMBER=${POSTGIS_VERSION_NUMBER})
 
-#--------------------------------
-
-
-add_definitions(-Wall -Wextra -std=gnu1x -Wno-unused-parameter)
```
Comment (cvvergara) [inline CMakeLists.txt:80]: Why remove these checks?
Remember a warning in one system might be an error in another

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,58 +44,58 @@ if(NOT WIN32)
...
-if (CMAKE_COMPILER_IS_GNUCC)
```
Comment (cvvergara) [inline CMakeLists.txt:81]: And your coverage stuff, why are you removing?

**Review by cvvergara — CHANGES_REQUESTED** [review]

**Comment by cvvergara** [comment]
This change is about generating the doxygen documentation
Formatting the file I don't see a problem but you are removing compilation flags.
And there are conflicts:
To update the working branch

If you haven't done this, its a one time thing:
```
git remote add upstream https://github.com/MobilityDB/MobilityDB
```
Now to put up-to-date  a branch: 
```
git fetch upstream
git rebase upstream/develop
```
If there are conflicts:
```
git status
```
Files with conflicts will be not staged, edit them to resolve the conflicts.
After fixing the conflicts:
```
git add <file>
git rebase --continue
```
and force the push
```
git push -f
```

File: CMakeLists.txt (hunk 2 above)
Comment (estebanzimanyi) [reply CMakeLists.txt:81]: I didn't remove anything, I just organized all compiler related issues into a single section entitled "Set compiler flags and definitions"

File: CMakeLists.txt (hunk 1 above)
Comment (estebanzimanyi) [reply CMakeLists.txt]: See comment below

**Review by cvvergara — APPROVED** [review]

**Pull request merged**